ghost/core/core/server/web/parent/middleware/queue-request.js

```diff
@@ -0,0 +1,76 @@
+const debug = require('@tryghost/debug')('middleware:queue-request');
+const metrics = require('@tryghost/metrics');
+const path = require('node:path');
+const expressQueue = require('express-queue');
+
+const DEFAULT_QUEUE_LIMIT = 100;
+
+module.exports = function queueRequest(
+    config,
+    queueFactory = expressQueue,
+    logMetric = metrics.metric.bind(metrics)
+) {
+    config = {
+        ...{
+            limit: DEFAULT_QUEUE_LIMIT
+        },
+        ...config
+    };
+
+    debug('Initialising middleware with config:', config);
+
+    // @see https://github.com/alykoshin/express-queue#usage
+    const queue = queueFactory({
+        activeLimit: config.limit,
+        queuedLimit: -1 // Do not limit the number of queued requests
+    });
+
+    /**
+     * Available events:
+     * - queue - when a request is queued
+     * - dequeue - when a request is dequeued
+     * - process - when a request is being processed
+     * - reject - when a request is rejected
+     * - cancel - when a request is cancelled
+     * - complete - when a request has completed
+     *
+     * @see https://github.com/search?q=repo:alykoshin/mini-queue%20job._toState&type=code
+     */
+
+    queue.queue.on('queue', (job) => {
+        debug(`Request queued: ${job.data.req.path}`);
+
+        logMetric('request-queue', {
```

Comment:
I do think logging the metrics is helpful, but one of the troubles without using `GhostLogger` is that we will be changing the definition of the `res.time` field in elastic. With the current approach here we aren't using the 'total time in Ghost' as the `res.time` value, just the time spent out of the queue. In order to get a good comparison I think we want to update the `req.startTime` and `res.responseTime` fields.

Reply:
Discussed on call, having `queueDepth` on the req is valuable for debugging. Have moved from metric to request in https://github.com/TryGhost/Ghost/pull/19887/commits/51da221ef812d693bf534d20d2ff359ce3e82703 👍